jekyll/_cci2/server/installation/migrate-from-server-2-to-server-4.adoc

```diff
@@ -0,0 +1,109 @@
+---
+contentTags:
+  platform:
+  - Server v2.x
+  - Server v4.x
+  - Server Admin
+---
+= CircleCI Server v2.19 to v4.0 Migration
+:page-layout: classic-docs
+:page-liquid:
+:page-description: Learn now to migrate from CircleCI server 2.19.x to 4.0.
+:icons: font
+:toc: macro
+:toc-title:
+
+Migrating from v2.19.x to v4.0 requires you to back up your v2.19 instance data (including Mongo, Postgres, and Vault), and then restore that data in a waiting server v4.0 instance. If you experience problems, you can fall back to your v2.19 instance. Migration requires an operational server v4.0 installation.
+
+Depending on the size of your data stores, the migration can take anywhere from a few minutes to a few hours. Using a staging environment before completing this process in a production environment is recommended. Using a staging environment not only allows you to gain a better understanding of the migration process, but also gives you an idea of how long the migration will take to complete.
+
+[#prerequisites]
+== Prerequisites
+
+. Your current CircleCI server installation is v2.19x
+. You have taken a backup of your v2.19 instance. If you are using external datastores, they need to be backed up separately.
+. If using externalized datastores, Postgres must be updated to version 12.
+. You have a new CircleCI server v4.0 link:/docs/server/installation/phase-1-prerequisites[installation].
+. You have successfully run link:https://github.com/circleci/realitycheck[realitycheck] with contexts before starting.
+. The migration script must be run from a machine with:
+- `kubectl` configured for the server v4.0 instance
+- `ssh` access to the v2.19 services machine
+
+[#migration]
+== Migration
+
+WARNING: Migrating to server v4.0 will shut down your v2.19 application. Your v2.19 application will not be restarted, although you may manually restart it using the KOTS Admin Console. Running server v2.19 and server v4.0 at the same time can cause issues with your 2.19 build data. Server v2.19 should NOT be restarted if server v4.0 is running.
```

Comment:
This part still references KOTS, needs updating I think?

Reply:
fixed!